src/pkg/bundle/publish.go

```diff
@@ -41,8 +43,38 @@ func (b *Bundle) Publish() error {
 		return err
 	}
 
-	// unarchive bundle into empty tmp dir
-	err = av3.Unarchive(b.cfg.PublishOpts.Source, b.tmp) // todo: awkward to use old version of mholt/archiver
+	// construct a fileHandler that extracts all files from the archive with the same relative pathing
+	fileHandler := func(_ context.Context, file archives.FileInfo) error {
+		extractPath := filepath.Join(b.tmp, file.NameInArchive)
+
+		if file.IsDir() {
+			return os.MkdirAll(extractPath, 0744)
+		}
+
+		if err = os.MkdirAll(filepath.Dir(extractPath), 0744); err != nil {
+			return err
+		}
+
+		stream, err := file.Open()
+		if err != nil {
+			return err
+		}
+		defer stream.Close()
+
+		fileBytes, err := io.ReadAll(stream)
+		if err != nil {
+			return err
+		}
+
+		return os.WriteFile(extractPath, fileBytes, 0644)
```

Comment:
Here and elsewhere;

I kind of just smashed against my numpad when setting file & directory permissions. We should probably use a tiny bit of brainpower to make sure the perms we set make sense.